distribution/lib/Standard/Test/0.0.0-dev/src/Main.enso

```diff
@@ -272,12 +272,16 @@ Error.should_equal _ frames_to_skip=0 = here.fail_match_on_unexpected_error this
          example_should_equal =
             1.00000001 . should_equal 1.00000002 epsilon=0.0001
 Decimal.should_equal : Decimal -> Decimal -> Integer -> Assertion
-Decimal.should_equal that (epsilon = 0) (frames_to_skip=0) = case this.equals that epsilon of
-    True -> Success
-    False ->
-        loc = Meta.get_source_location 2+frames_to_skip
-        msg = this.to_text + " did not equal " + that.to_text + " (at " + loc + ")."
-        Panic.throw (Failure msg)
+Decimal.should_equal that (epsilon = 0) (frames_to_skip=0) =
+    matches = case that of
+        Number -> this.equals that epsilon
+        _ -> False
+    case matches of
+        True -> Success
+        False ->
+            loc = Meta.get_source_location 2+frames_to_skip
```

Comment:
I know this was here before but where does the magic `2` come from?

Reply:
The number of frames of how deep inside of `should_equal` we are. I think the `case` itself adds one frame and then the particular branch another - was experimentally measured to give the correct result. Maybe worth double-checking.

Ideally we should have tests for this - but currently we do not have a robust way to test the test suite using the test suite, I think :D